packages/flutter/lib/src/widgets/_web_browser_detection_io.dart

```diff
@@ -0,0 +1,12 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// The widget in this file is an empty mock for non-web platforms. See
+// `_web_browser_detection.dart` for the web implementation.
+
+/// A class that provides information about the current browser when on the web.
+class WebBrowserDetection {
+  /// Whether the current browser is webkit (Safari).
+  static bool get browserIsSafari => throw UnimplementedError();
```

Comment:
What happens if you try to call `ui_web.BrowserDetection.instance.isSafari` on a non-web platform?

Reply:
You can't import `ui_web` on a non-web platform, you get a compile time error.

```
error: ../../flutter/packages/flutter/lib/src/widgets/editable_text.dart:20:8: Error: Dart library 'dart:ui_web' is not available on this platform.
import 'dart:ui_web' as ui_web;
       ^
Context: The unavailable library 'dart:ui_web' is imported through these packages:

    package:selectionplayground => package:flutter => dart:ui_web
```